### PR TITLE
[FIX] Regression Tree Sparse Support

### PR DESCRIPTION
**Pull request by jerneju**
##### Issue
Regression Tree Sparse Support

##### Includes
- [X] Code changes
- [x] Tests
- [ ] Documentation

### CODE REVIEW

**Comment by nikicc** [comment]
Continuation of #2370.

File: Orange/regression/tree.py [hunk 1]
```diff
@@ -95,8 +96,11 @@ def _score_disc_bin():
             # we don't do it here
             if score == 0:
                 return REJECT_ATTRIBUTE
+            datax = data.X[:, attr_no]
+            if is_sparse:
+                datax = datax.toarray()
             mapping, branches = MappedDiscreteNode.branches_from_mapping(
-                data.X[:, attr_no], mapping, len(attr.values))
+                datax, mapping, len(attr.values))
```
Comment (nikicc) [inline Orange/regression/tree.py]: Couldn't we use `col_x` here?

File: Orange/widgets/model/tests/test_tree.py [hunk 2]
```diff
@@ -41,17 +41,32 @@ def test_parameters_unchecked(self):
...
+        """
+        table1 = Table("housing")
+        self.send_signal("Data", table1)
+        model_dense = self.get_output("Model")
+        table2 = Table("housing")
+        table2.X = sp.csr_matrix(table2.X)
+        self.send_signal("Data", table2)
         model_sparse = self.get_output("Model")
         self.assertTrue(np.array_equal(model_dense._code, model_sparse._code))
-        self.assertTrue(np.array_equal(model_dense._thresholds, model_sparse._thresholds))
```
Comment (nikicc) [inline Orange/widgets/model/tests/test_tree.py]: Why is this removed from `test_sparse_data_classification `? Is this not relevant for regression?

Comment (jerneju) [inline Orange/widgets/model/tests/test_tree.py]: Because it fails.

Comment (jerneju) [inline Orange/widgets/model/tests/test_tree.py]: And I think this is fixed in #2496.

Comment (nikicc) [inline Orange/widgets/model/tests/test_tree.py]: But it didn't fail before for classification, if it was among tests, right?

Comment (jerneju) [inline Orange/widgets/model/tests/test_tree.py]: Before did not fail since the test did not run properly. Sparse table had never been sent to the widget.

Comment (nikicc) [inline Orange/widgets/model/tests/test_tree.py]: I see.
